frontstage/controllers/party_controller.py

```diff
@@ -154,3 +175,25 @@ def verify_token(token):
         raise ApiError(logger, response, log_level=log_level, message='Failed to verify token', token=token)
 
     logger.debug('Successfully verified token')
+
+
+def get_enrolment_details(business_party_id, enrolments):
+    business_details = get_party_by_business_id(business_party_id)
+    enrolments_with_details = [{"business_party": business_details, "enrolment_details": enrolment}
+                               for enrolment in enrolments
+                               if enrolment['enrolmentStatus'] == 'ENABLED']
+
+    return enrolments_with_details
+
+
+def get_survey_list_details_for_party(party_id, tag):
+    logger.debug("Get party enrolments", party_id=party_id)
+
+    respondent = get_respondent_party_by_id(party_id)
+    enrolments = [get_enrolment_details(association['partyId'], association['enrolments'])
+                  for association in respondent['associations']]
+    flattened_enrolments = [enrolment for sublist in enrolments for enrolment in sublist]
```

Comment:
This might be cleaner if we have a `get_respondent_enrolments(respondent)` function that runs these two lines.

Then this function would read nicely without any list comprehensions
```
respondent = get_respondent_party_by_id(party_id)
enrolments = get_respondent_enrolments(respondent)
enrolments_with_surveys = survey_controller.get_surveys_with_enrolments(flattened_enrolments)
enrolments_with_ces = collection_exercise_controller.get_enrolments_with_collection_exercises(enrolments_with_surveys)
enrolments_with_cases = case_controller.get_enrolments_with_cases(enrolments_with_ces, tag)
```